scheduler/src/cook/mesos/util.clj

```diff
@@ -375,16 +389,32 @@
 (timers/deftimer [cook-mesos scheduler get-running-tasks-duration])
 
 (defn get-running-task-ents
-  "Returns all running task entities"
+  "Returns all running task entities."
   [db]
   (timers/time!
     get-running-tasks-duration
-    (->> (q '[:find ?i
+    (->> (q '[:find [?i ...]
               :in $ [?status ...]
               :where
               [?i :instance/status ?status]]
             db [:instance.status/running :instance.status/unknown])
-         (map (fn [[x]] (d/entity db x))))))
+         (map (partial d/entity db)))))
+
+(timers/deftimer [cook-mesos scheduler get-user-running-tasks-duration])
+
+(defn get-user-running-task-ents
+  "Returns all running task entities for a specific user."
+  [db user]
+  (timers/time!
+    get-user-running-tasks-duration
+    (->> (q '[:find [?i ...]
+              :in $ [?status ...] ?user
+              :where
+              [?j :job/user ?user]
```

Comment:
I think this query will be more performant if we put the `:job/status` condition before the `:job/user` condition. Some users will have a couple of orders of magnitude more completed jobs than currently running jobs. In any case it's worth testing the query out first.

Reply:
Using `:job/state` here actually simplified a lot of other stuff too. Thanks for the suggestion!